File: week3/Eren-Taskiran/153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int right=nums.length-1;
        int left = 0;
        while(left<right){
            final int mid=(right+left)/2;
            if(nums[mid]<nums[right]){
                right=mid;
            }
            else left = mid+1;
        }
        return nums[left];
    }
}

File: week3/atakantekatan/153. Find Minimum in Rotated Sorted Array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums)-1
        m = float("inf")
        while l <= r:
            mid = (l+r) // 2
            m = min(m, nums[mid])
            
            if nums[mid] > nums[r]:
                if nums[l] < nums[r]:
                    r = mid - 1
                else:
                    l = mid + 1
            else:
                r = mid - 1
        return m